sdks/python/apache_beam/io/localfilesystem.py

```diff
@@ -127,6 +127,9 @@ def _path_open(self, path, mode, mime_type='application/octet-stream',
     """Helper functions to open a file in the provided mode.
     """
     compression_type = FileSystem._get_compression_type(path, compression_type)
+    parent = os.path.dirname(path)
```

Comment:
We should only create the path in the `create` call and not the `open` call as we'll get a weird error if the user mistypes the path for something being read and we will try to create the directory which may fail (e.g. permissions) which will raise a confusing error message.

Do you want to add a test to localfilesystem_test.py so that this isn't regressed?